Client/TabManagement/TabManagerImplementation.swift

```diff
@@ -363,39 +365,40 @@ class TabManagerImplementation: LegacyTabManager, Notifiable {
         return false
     }
 
-    // MARK: - Save screenshot
-    override func tabDidSetScreenshot(_ tab: Tab, hasHomeScreenshot: Bool) {
-        guard shouldUseNewTabStore()
-        else {
-            super.tabDidSetScreenshot(tab, hasHomeScreenshot: hasHomeScreenshot)
+    // MARK: - Screenshots
+
+    func tabDidSetScreenshot(_ tab: Tab, hasHomeScreenshot: Bool) {
+        guard tab.screenshot != nil else {
+            // Remove screenshot from image store so we can use favicon
+            // when a screenshot isn't available for the associated tab url
+            removeScreenshot(tab: tab)
             return
         }
 
         storeScreenshot(tab: tab)
     }
 
-    override func storeScreenshot(tab: Tab) {
-        guard shouldUseNewTabStore(),
-              let screenshot = tab.screenshot
-        else {
-            super.storeScreenshot(tab: tab)
-            return
-        }
+    func storeScreenshot(tab: Tab) {
+        guard let screenshot = tab.screenshot else { return }
 
         Task {
             try? await imageStore?.saveImageForKey(tab.tabUUID, image: screenshot)
         }
     }
 
-    override func removeScreenshot(tab: Tab) {
-        guard shouldUseNewTabStore()
-        else {
-            super.removeScreenshot(tab: tab)
-            return
+    func removeScreenshot(tab: Tab) {
+        Task {
+            await imageStore?.deleteImageForKey(tab.tabUUID)
         }
+    }
 
+    func cleanUpUnusedScreenshots() {
```

Comment:
Should we test this?

Reply:
Yes, it should be called as a byproduct of the restore function so should be tested as part of those tests but they are disabled right now due to being flaky. This function should be private, I'll fix it in my next PR. I'll also take a look at reinstating the flaky tests before I move on from this task.